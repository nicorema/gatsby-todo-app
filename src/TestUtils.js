import React from 'react'
import { IntlContextProvider, IntlProvider } from 'gatsby-plugin-intl'

export const renderWithProviders = component => (
  <IntlContextProvider value={{ language: 'en-US', routed: true }}>
    <IntlProvider locale="en">{component}</IntlProvider>
  </IntlContextProvider>
)
