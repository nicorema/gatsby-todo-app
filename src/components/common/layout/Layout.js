import React from 'react'
import Navigation from '../../navigation/Navigation'
import Button from '../button/Button'
import { StyledLayout, StyledHeader, StyledFooter } from './Layout.styles'

const Layout = ({ children, stickyAction, title }) => (
  <StyledLayout>
    <StyledHeader>
      <h1>{title}</h1>
      {stickyAction && <Button {...stickyAction} />}
    </StyledHeader>
    {children}
    <StyledFooter>
      <Navigation />
    </StyledFooter>
  </StyledLayout>
)

export default Layout
