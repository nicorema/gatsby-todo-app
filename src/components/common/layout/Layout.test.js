import React from 'react'
import renderer from 'react-test-renderer'
import { renderWithProviders } from '../../../TestUtils'

import Layout from './Layout'

describe('Layout', () => {
  const stickyAction = {
    onClick: jest.fn(),
  }
  const layoutCmp = renderer.create(
    renderWithProviders(
      <Layout title="test title" stickyAction={stickyAction}>
        <p>Testing child</p>
      </Layout>
    )
  )

  it('renders correctly', () => {
    const snapShot = layoutCmp.toJSON()
    expect(snapShot).toMatchSnapshot()
  })

  it('executes stickyAction on click', () => {
    layoutCmp.root.findByType('button').props.onClick()
    expect(stickyAction.onClick).toHaveBeenCalled()
  })

  it('does not render button when there is no stickyAction', () => {
    layoutCmp.update(
      renderWithProviders(
        <Layout title="test title">
          <p>Testing child</p>
        </Layout>
      )
    )
    const button = layoutCmp.root.findAllByType('button')
    expect(button).toHaveLength(0)
  })
})
