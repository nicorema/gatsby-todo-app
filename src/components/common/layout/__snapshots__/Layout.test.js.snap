// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="css-1mjppp"
>
  <header
    className="sc-dlnjPT eSGkCO"
  >
    <h1>
      test title
    </h1>
    <div
      className="sc-gtssRu loStDX"
    >
      <button
        onClick={[MockFunction]}
      />
    </div>
  </header>
  <p>
    Testing child
  </p>
  <footer
    className="css-1ty9u91"
  >
    <nav
      className="sc-bdnylx giuwNz"
    >
      <ul>
        <li>
          <span
            className="css-qyfojd"
          >
            <a
              href="/en-US/todos-manager"
              onClick={[Function]}
            >
              Todos Manager
            </a>
          </span>
        </li>
        <li>
          <span
            className="css-qyfojd"
          >
            <a
              href="/en-US/todos-log"
              onClick={[Function]}
            >
              Todos Log
            </a>
          </span>
        </li>
      </ul>
    </nav>
  </footer>
</div>
`;
